File: presentation-examples/inheritance.java
import sqlx.connection.PgConnection;
import reqwest.Client;

public class Client implements From<PgConnection> {
    private Client reqwest;
    private PgConnection conn;

    public Client (PgConnection conn) {
        this.reqwest = new reqwest.Client();
        this.conn = conn;
    };

    public Client from(PgConnection value) {
        return new Client(value);
    }
}

File: presentation-examples/inheritance.rs
use sqlx::connection::PgConnection;

pub struct Client {
    reqwest: reqwest::Client,
    conn: PgConnection,
}

impl From<PgConnection> for Client {
    fn from(value: PgConnection) -> Client {
        Client {
            reqwest: reqwest::Client::default(),
            conn: value,
        }
    }
}
